File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> temp = new HashSet<>(); 
        for (int num : nums){
            temp.add(num);
        }
        int longest = 0;
        for (int num: temp){
            if (!temp.contains(num-1)){
                int length = 1;
                while(temp.contains(num + length)){
                    length++;
                }
                longest = Math.max(length, longest);
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        temp = set()
        maxLen = 0
        l, r = 0, 0
        while r < len(s):
            if s[r] not in temp:
                temp.add(s[r])
                maxLen = max(maxLen, r - l + 1)
                r += 1
            else:
                temp.remove(s[l])
                l += 1
        return maxLen

